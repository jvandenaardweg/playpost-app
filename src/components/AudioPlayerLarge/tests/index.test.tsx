import React from 'react';

// tslint:disable-next-line: no-submodule-imports
import * as ShallowRenderer from 'react-test-renderer/shallow';

import articleMock from '../../../../tests/__mocks__/article';
import { AudioPlayerLarge, Props } from '../index';

const onSetPlaybackSpeedHandler = jest.fn();
const onPressPlayHandler = jest.fn();
const onProgressChangeHandler = jest.fn();
const onTogglePlaybackSpeedVisibilityHandler = jest.fn();
const onPressJumpForwardHandler = jest.fn();
const onPressJumpBackwardHandler = jest.fn();
const onPressVoiceHandler = jest.fn();

const defaultProps: Props = {
  onSetPlaybackSpeed: onSetPlaybackSpeedHandler,
  onPressPlay: onPressPlayHandler,
  onProgressChange: onProgressChangeHandler,
  onTogglePlaybackSpeedVisibility: onTogglePlaybackSpeedVisibilityHandler,
  onPressJumpForward: onPressJumpForwardHandler,
  onPressJumpBackward: onPressJumpBackwardHandler,
  onPressVoice: onPressVoiceHandler,
  playbackSpeed: 1,
  isPlaying: false,
  isLoading: false,
  isPlaybackSpeedVisible: false,
  article: articleMock
}

describe('AudioPlayerLarge', () => {
  describe('rendering', () => {
    let wrapper: any;

    const props = {
      ...defaultProps
    }

    beforeAll(() => {
      const renderer = ShallowRenderer.createRenderer();
      renderer.render(<AudioPlayerLarge {...props} />);
      wrapper = renderer.getRenderOutput();
    });

    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  // describe('interacting', () => {
  //   let wrapper: RenderAPI;

  //   const props = {
  //     ...defaultProps
  //   }

  //   beforeEach(() => {
  //     jest.resetAllMocks();
  //     jest.restoreAllMocks();
  //     wrapper = render(<PlaybackSpeedSlider {...props} />);
  //   });

  //   it('should up the playback speed with 0.01 when pressing the plus button', () => {
  //     fireEvent.press(wrapper.getByTestId('PlaybackSpeedSlider-TouchableHighlight-plus'));
  //     expect(onSetPlaybackSpeedHandler).toHaveBeenCalledTimes(1);
  //     expect(onSetPlaybackSpeedHandler).toHaveBeenCalledWith(1.01);
  //   });

  //   it('should down the playback speed with 0.01 when pressing the minus button', () => {
  //     fireEvent.press(wrapper.getByTestId('PlaybackSpeedSlider-TouchableHighlight-minus'));
  //     expect(onSetPlaybackSpeedHandler).toHaveBeenCalledTimes(1);
  //     expect(onSetPlaybackSpeedHandler).toHaveBeenCalledWith(0.99);
  //   });

  //   it('should reset the playback speed to default when pressing the label', () => {
  //     // Press the plus once
  //     fireEvent.press(wrapper.getByTestId('PlaybackSpeedSlider-TouchableHighlight-plus'));
  //     expect(onSetPlaybackSpeedHandler).toHaveBeenCalledWith(1.01);

  //     // Reset
  //     fireEvent.press(wrapper.getByTestId('PlaybackSpeedSlider-Text-label'));
  //     expect(onSetPlaybackSpeedHandler).toHaveBeenCalledWith(1);
  //   });
  // });
});
