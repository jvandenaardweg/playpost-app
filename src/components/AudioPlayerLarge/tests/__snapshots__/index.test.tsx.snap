// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AudioPlayerLarge rendering should render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#000000",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#1f1f1f",
          "borderBottomColor": "#333333",
          "borderBottomWidth": 1,
          "borderTopColor": "#333333",
          "borderTopWidth": 1,
          "flexGrow": 1,
          "flexShrink": 1,
          "marginBottom": 20,
          "marginTop": 0,
          "overflow": "hidden",
        }
      }
    >
      <Memo
        article={
          Object {
            "audiofiles": Array [
              Object {
                "bucket": "storage-development.playpost.app",
                "createdAt": "2019-08-16T10:32:32.821Z",
                "filename": "articles/7ca86b8a-02aa-4733-92e4-1cfe7ef46954/audiofiles/2f495b91-bb7f-449e-8333-45b9d4905463.mp3",
                "id": "2f495b91-bb7f-449e-8333-45b9d4905463",
                "length": 252.12,
                "mimeType": "audio/mpeg",
                "updatedAt": "2019-08-16T10:32:32.821Z",
                "url": "https://storage-development.playpost.app/articles/7ca86b8a-02aa-4733-92e4-1cfe7ef46954/audiofiles/2f495b91-bb7f-449e-8333-45b9d4905463.mp3",
                "voice": Object {
                  "country": Object {
                    "code": "US",
                    "continent": "NA",
                    "createdAt": "2019-08-15T12:26:33.233Z",
                    "currency": "USD,USN,USS",
                    "id": "3bc87ff6-f1c1-4bb7-b8b3-22555e07957b",
                    "name": "United States",
                    "nativeName": "United States",
                    "updatedAt": "2019-08-15T12:26:33.233Z",
                  },
                  "countryCode": "US",
                  "createdAt": "2019-04-25T07:27:13.250Z",
                  "exampleAudioUrl": "https://storage-development.playpost.app/voices/3675e6db-307e-48de-b5ff-13b38470632f.wav",
                  "gender": "FEMALE",
                  "id": "3675e6db-307e-48de-b5ff-13b38470632f",
                  "isActive": true,
                  "isLanguageDefault": null,
                  "isPremium": true,
                  "isSubscribedLanguageDefault": true,
                  "isUnsubscribedLanguageDefault": null,
                  "label": "Emily",
                  "language": Object {
                    "code": "en",
                    "createdAt": "2019-05-16T12:51:57.546Z",
                    "id": "95f01039-fe8e-412d-a542-7089303b84c8",
                    "isActive": true,
                    "name": "English",
                    "nativeName": "English",
                    "rightToLeft": false,
                    "updatedAt": "2019-08-15T13:28:48.797Z",
                  },
                  "languageCode": "en-US",
                  "name": "en-US-Wavenet-F",
                  "quality": "Very High",
                  "synthesizer": "Google",
                  "updatedAt": "2019-09-17T10:37:30.995Z",
                },
              },
            ],
            "authorName": "Darius Foroux",
            "canonicalUrl": "https://forge.medium.com/how-to-keep-going-if-you-dont-know-what-s-next-9cb8346885f7",
            "compatibilityMessage": "All checks passed successfully! It is probably an article.",
            "createdAt": "2019-08-16T10:28:55.551Z",
            "description": "THE FUTURE CAN BE SCARY, BUT THERE ARE WAYS TO DEAL WITH THAT FEAR We often hear about the power of day one, and are told that getting started is the key to success. But really, the hardest and most...",
            "html": "<div id=\\"page-1\\" class=\\"page\\"><div><div><h2>The future can be scary, but there are ways to deal with that fear</h2><div><div><div><a href=\\"https://link.medium.com/@dariusforoux?source=post_page-----9cb8346885f7----------------------\\"><p><img src=\\"https://miro.medium.com/fit/c/96/96/1*njIyBnA2NDsf1toYE1rfQw.jpeg\\"></p></a></div></div></div><figure><div><div><div><p><img src=\\"https://miro.medium.com/max/60/1*G32Y6J9RapeGziq7ajRa9A.jpeg?q=20\\"></p><p><img></p></div></div></div><figcaption>Photo: Hero Images/Getty Images</figcaption></figure><p><span><img src=\\"https://miro.medium.com/max/158/1*xS07uldAkOcvvOet1aMRvQ.png\\"><span>We</span></span> often hear about the power of day one, and are told that getting started is the key to success. But really, the hardest and most vital thing is to keep going.</p><p>Anyone can declare themselves a writer, singer, designer, illustrator, and many have real talent. But only a dedicated few can stick with it for months, years, decades. Plenty of people have great ideas and entrepreneurial spirit. But <a href=\\"https://www.bls.gov/bdm/us_age_naics_00_table7.txt\\">almost half of startups</a> don’t exist after five years.</p><p>A reader of my blog recently emailed me: “I want to create something for others,” the person wrote. “I started with a tiny step. It was easy at first. But after that I felt like I lacked inspiration. How do you keep going even if you don’t know what’s next?”</p><p>It’s an essential question. Not knowing what’s next is a fundamental condition of human life. Here’s what I do to tackle that challenge.</p><h2>Focus on today</h2><p>Much of our anxiety comes from uncertainty about the future. “What will happen tomorrow or in a year from now?” It can be scary. I can tell you that the world won’t blow up tomorrow, but if you’re in the wrong state of mind, you won’t listen to a word I say.</p><p>The best thing you can do is to focus on your very next step — right now. What are you going to do next? Not as in, “What’s your next big move?” No. What are you going to do after you’ve read this article?</p><p>Look, maybe you’re right. Maybe tomorrow the world will end. Are you really okay with wasting today on crap you don’t control? Make something out of your day. Forget about the past and the future.</p><h2>Set goals in areas you control</h2><p>Never give the wheel to someone else.<a href=\\"https://dariusforoux.com/be-free/\\"> You’re the ruler of your kingdom.</a> You’re in the driver’s seat.</p><p>That’s why you’ll want to focus on things you can control. You don’t control the rewards you get. If people don’t want to give you their business, so what? If people don’t want to hire you, so what? When you set goals you don’t have control over, you will think success is impossible. Plus, you will always depend on others for your success.</p><p>Your business is to improve yourself, make things, be a good person, and do the best you can.</p><p>Think about what you actually control. Very few things, right? Among them: your own work, your learning, your openness to new ideas. That should make this step easy. All you have to do is to grab a piece of paper and write your goals on it. Next, work on them.</p><h2>Adjust — or even bail — when it’s necessary</h2><p>A while back, the author and podcaster Tim Ferriss tried to change the business model for his podcast. He went from ad-based to fan-based funding. He asked listeners to contribute $10 a month, or more if they wanted. He announced that this was an experiment for something like six months.</p><p>Then, a few weeks later, he pulled the plug, refunded all the contributions, and went back to ads. That’s life. No big deal. He didn’t stop podcasting. He didn’t become a failure. He simply made an adjustment.</p><p>Ferriss put a positive spin on it, pointing out that “experiments can sometimes yield conclusions much more quickly than expected.”</p><p>And he’s right — if we are paying attention. Sometimes we try something new, and we soon find that it’s not going according to plan. Perhaps our assumptions and plans were totally off, or perhaps circumstances changed. Either way,<a href=\\"https://dariusforoux.com/better-decisions/\\"> we should not be afraid to make bold decisions</a>.</p><blockquote><p>Keep turning the knobs, my friend. When it sounds right, you’ll know.</p></blockquote><p>These steps are key to finding the right fit for your career and life: focusing on the day ahead of you, setting goals in areas you can control, and making adjustments where necessary.</p><p>It reminds me of what a music producer once told me when I asked him about his creative process: “I come up with a melody, turn it into a song, and then finally, I keep turning the knobs until it sounds right.”</p><p>Keep turning the knobs, my friend. When it sounds right, you’ll know.</p></div></div></div>",
            "id": "7ca86b8a-02aa-4733-92e4-1cfe7ef46954",
            "imageUrl": "https://miro.medium.com/focal/1200/632/77/62/1*G32Y6J9RapeGziq7ajRa9A.jpeg",
            "isCompatible": true,
            "language": Object {
              "code": "en",
              "createdAt": "2019-05-16T12:51:57.546Z",
              "id": "95f01039-fe8e-412d-a542-7089303b84c8",
              "isActive": true,
              "name": "English",
              "nativeName": "English",
              "rightToLeft": false,
              "updatedAt": "2019-08-15T13:28:48.797Z",
            },
            "readingTime": 215.70000000000002,
            "sourceName": "Medium",
            "status": "finished",
            "title": "How to Keep Going When You Don’t Know What’s Next",
            "updatedAt": "2019-08-16T10:29:01.954Z",
            "url": "https://link.medium.com/XhWppETqcZ",
          }
        }
        theme="dark"
      />
    </View>
    <View
      style={
        Object {
          "paddingLeft": 20,
          "paddingRight": 20,
        }
      }
    >
      <Memo
        onProgressChange={[MockFunction]}
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <TouchableOpacity
          activeOpacity={0.2}
          hitSlop={
            Object {
              "bottom": 5,
              "left": 5,
              "right": 5,
              "top": 5,
            }
          }
          onPress={[MockFunction]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#444444",
              "borderRadius": 32,
              "height": 32,
              "justifyContent": "center",
              "marginLeft": 8,
              "marginRight": 8,
              "marginTop": 12,
              "paddingLeft": 14,
              "paddingRight": 14,
              "width": 65,
            }
          }
          testID="AudioPlayerLarge-TouchableOpacity-playbackspeed"
        >
          <Memo
            style={
              Object {
                "color": "#ffffff",
              }
            }
            template="footnoteEmphasized"
            testID="AudioPlayerLarge-Text-playbackspeed"
          >
            1.00
            x
          </Memo>
        </TouchableOpacity>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-around",
            "marginTop": 20,
          }
        }
      >
        <View
          style={
            Object {
              "width": 48,
            }
          }
        >
          <TouchableOpacity
            activeOpacity={0.2}
            hitSlop={
              Object {
                "bottom": 20,
                "left": 20,
                "right": 20,
                "top": 20,
              }
            }
            onPress={[MockFunction]}
            testID="AudioPlayerLarge-TouchableOpacity-jump-backward"
          >
            <Memo
              color="white"
              name="replay-10"
              size={42}
            />
          </TouchableOpacity>
        </View>
        <View
          style={
            Object {
              "height": 64,
              "opacity": 1,
            }
          }
        >
          <Memo
            iconColor="#000000"
            isDisabled={false}
            isLoading={false}
            isPlaying={false}
            onPressPlay={[MockFunction]}
            size={24}
          />
        </View>
        <View>
          <View
            style={
              Object {
                "width": 48,
              }
            }
          >
            <TouchableOpacity
              activeOpacity={0.2}
              hitSlop={
                Object {
                  "bottom": 20,
                  "left": 20,
                  "right": 20,
                  "top": 20,
                }
              }
              onPress={[MockFunction]}
              testID="AudioPlayerLarge-TouchableOpacity-jump-forward"
            >
              <Memo
                color="white"
                name="forward-10"
                size={42}
              />
            </TouchableOpacity>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "height": 60,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 12,
        }
      }
    >
      <AnimatedComponent
        style={
          Array [
            Object {
              "flexGrow": 1,
              "width": "100%",
            },
            Object {
              "transform": Array [
                Object {
                  "translateY": 150,
                },
              ],
            },
          ]
        }
      >
        <Memo
          onSetPlaybackSpeed={[MockFunction]}
          playbackSpeed={1}
        />
      </AnimatedComponent>
    </View>
  </View>
</View>
`;
